### DISCUSSION TITLE
what about custom roms loader?

### DISCUSSION THREAD

**Discussion by koswir** [eh2k/squares-and-circles#105] [Ideas]
Hey @eh2k, I was thinking about loading my own samples into S&C, and it reminded me that you already did the 707 ROM loader. While browsing, I found this: [Drumulator ROM Creator on Matrixsynth](https://www.matrixsynth.com/2019/08/emu-drumulator-sound-rom-creator.html) - a tool for creating Drumulator sound ROMs.

What if you made a Drumulator ROM loader with a web interface to upload custom ROMs? That way, people could easily build their own sample sets. What do you think?

**Comment by eh2k** [comment]
@koswir 

I took a look at the drumulator files from your link.

The four ROMs are exactly 64kb in total, and contain a set of samples (i'm not sure how exactly the samples are addressed, presumably the addresses are in the separate os rom). 
The [mu-law](
https://en.wikipedia.org/wiki/%CE%9C-law_algorithm) samples are basically 8bit and very memory friendly similar to the TR707 ROM (also 64kb).

I also like the vintage samples from the 80s. However, I try to keep them out of the firmware for licence reasons.
The TR707 or TR909 engines do not contain the sample data, but reference the ROMs that the user has to copy to the SD card (or flash to the device). It would actually be the same here. 

Here is a very nice website where you can find many original roms. If you have something specific with low memory requirements I could take a closer look.

https://dbwbp.com/index.php/9-misc/37-synth-eprom-dumps

I already have the LinDrum or DMX on my to-do list. But the Drumulator wouldn't be bad as a general solution.

There was once a similar discussion, see here: https://github.com/eh2k/squares-and-circles/discussions/106#discussioncomment-11181630

**Reply by koswir** [reply]
Thank you for the response! 
I mentioned the Drumulator ROM loader concept primarily to make it easier for you, cause there is the TR-707 ROM loader already present in the software. 

As a user, I would just love to upload samples to the S&C. I thought the Drumulator ROM loader might an easier walkaround and users could create custom, compatible ROMs. 

The ideal solution, though, would be a sampler that allows for user-uploaded samples through an interface. That would be a fantastic addition for everyone!

I’ve read the other topic and of course any classic drum machine set would be lovely too. I love DMX and Linn sound!
